src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -330,69 +367,127 @@ getUserProfile = getProfile >>= maybe noProfile pure
+    pure $ CWalletSet cAddr meta walletsNum
+  where
+    noWSet = throwM . Internal $
+        sformat ("No wallet set with address "%build%" found") cAddr
 
 -- TODO: probably poor naming
 decodeCAddressOrFail :: WalletWebMode ssc m => CAddress -> m Address
 decodeCAddressOrFail = either wrongAddress pure . cAddressToAddress
   where wrongAddress err = throwM . Internal $
             sformat ("Error while decoding CAddress: "%stext) err
 
-getWallets :: WalletWebMode ssc m => m [CWallet]
-getWallets = join $ mapM getWallet <$> myCAddresses
+getWallets :: WalletWebMode ssc m => Maybe CWalletSetAddress -> m [CWallet]
+getWallets mCAddr = do
+    whenJust mCAddr $ \cAddr -> getWSetMeta cAddr `whenNothingM_` noWSet cAddr
+    mapM getWallet . filterByAddr mCAddr =<< getWalletAddresses
+  where
+    filterByAddr = maybe identity $ \cAddr ->
+        filter $ (== cAddr) . cwaWSAddress
+    noWSet cAddr = throwM . Internal $
+        sformat ("No wallet set with address "%build%" found") cAddr
+
+getWSets :: WalletWebMode ssc m => m [CWalletSet]
+getWSets = getWSetAddresses >>= mapM getWSet
+
+decodeCPassPhraseOrFail :: WalletWebMode ssc m => CPassPhrase -> m PassPhrase
+decodeCPassPhraseOrFail cpass =
+    either (const . throwM $ Internal "Decoding of passphrase failed") return $
+    cPassPhraseToPassPhrase cpass
 
-send :: WalletWebMode ssc m => SendActions m -> CPassPhrase -> CAddress -> CAddress -> Coin -> m CTx
+send :: WalletWebMode ssc m => SendActions m -> CPassPhrase -> CAccountAddress -> CAccountAddress -> Coin -> m CTx
 send sendActions cpass srcCAddr dstCAddr c =
     sendExtended sendActions cpass srcCAddr dstCAddr c ADA mempty mempty
 
-sendExtended :: WalletWebMode ssc m => SendActions m -> CPassPhrase -> CAddress -> CAddress -> Coin -> CCurrency -> Text -> Text -> m CTx
+-- TODO [CSM-185]: Send between wallets, not accounts
+sendExtended :: WalletWebMode ssc m => SendActions m -> CPassPhrase -> CAccountAddress -> CAccountAddress -> Coin -> CCurrency -> Text -> Text -> m CTx
 sendExtended sendActions cpassphrase srcCAddr dstCAddr c curr title desc = do
     passphrase <- decodeCPassPhraseOrFail cpassphrase
-    srcAddr <- decodeCAddressOrFail srcCAddr
-    dstAddr <- decodeCAddressOrFail dstCAddr
-    idx <- getAddrIdx srcAddr
-    sks <- getSecretKeys
-    let sk = sks !! idx
+    let srcWCAddr = walletAddrByAccount srcCAddr
+    srcAddr <- decodeCAddressOrFail $ caaAddress srcCAddr
+    dstAddr <- decodeCAddressOrFail $ caaAddress dstCAddr
+    sk      <- getSKByAccAddr passphrase srcCAddr
+    let mainTx = TxOutAux (TxOut dstAddr c) []
+    balance <- getAccountBalance srcCAddr
+    mRems <- if balance < c
+        then return Nothing
+        else do
+            remCAddr <- caAddress <$> newAccount cpassphrase srcWCAddr
+            remAddr  <- decodeCAddressOrFail $ caaAddress remCAddr
+            let remTx = TxOutAux (TxOut remAddr $ balance `unsafeSubCoin` c) []
+            return $ Just (remTx, remCAddr)
```

Comment:
1. Seems like it should be `if balance <= c`
2. As I stated before, there should be a way to gather funds for transaction from different addresses in the wallet pool. Now it's not done in that way. We probably can leave it like this as a temp solution, but it should be rewritten soon.